ddtrace/internal/compat.py

```diff
@@ -117,6 +117,14 @@ def process_time_ns():
         return int(_process_time() * 1e9)
 
 
+# typing_extensions backports TypedDict for python versions < 3.8
+# this type is used in ddtrace/internal/telemetry
+if sys.version_info < (3, 8):
+    from typing_extensions import TypedDict
+else:
+    from typing import TypedDict
+TypedDict = TypedDict
```

Comment:
I tried to follow the existing patten using the example shown below but this threw an exception in mypy  
```
if sys.version_info < (3, 8):
    TypedDict = typing.TypedDict
else:
  import typing_extensions
  TypedDict = typing.TypedDict
```